src/backends/torch/torchlib.cc

```diff
@@ -163,6 +163,84 @@ namespace dd
       _classif->to(device, dtype);
   }
 
+  void TorchModule::proto_model_load(const TorchModel &model)
+  {
+    _logger->info("loading " + model._proto);
+    try
+      {
+        _graph = std::make_shared<CaffeToTorch>(model._proto);
+      }
+    catch (std::exception &e)
+      {
+        _logger->info("unable to load " + model._proto);
+        throw;
+      }
+  }
+
+  void TorchModule::graph_model_load(const TorchModel &tmodel)
+  {
+    if (!tmodel._traced.empty() && _graph->needs_reload())
+      {
+        _logger->info("loading " + tmodel._traced);
+        try
+          {
+            torch::load(_graph, tmodel._traced, _device);
+          }
+        catch (std::exception &e)
+          {
+            _logger->error("unable to load " + tmodel._traced);
+            throw;
+          }
+      }
+  }
+
+  void TorchModule::native_model_load(const TorchModel &tmodel,
+                                      const torch::Device device)
+  {
+    if (!tmodel._native.empty())
+      {
+        _logger->info("loading " + tmodel._native);
+        try
+          {
+            torch::load(_native, tmodel._native, device);
+          }
+        catch (std::exception &e)
+          {
+            _logger->error("unable to load " + tmodel._native);
+            throw;
+          }
+      }
+  }
+
+  void TorchModule::classif_model_load(const TorchModel &model)
+  {
+    _logger->info("loading " + model._weights);
+    try
+      {
+        torch::load(_classif, model._weights, _device);
```

Comment:
This seems to me a generic loader, from `_model._weight`, I guess this is called `_classif` becauses it started by supporting classification ?

Additionnally, there's no support for the `device` parameter for that type of model ?

Reply:
_classif is a classification head added on top of traced model for classification tasks. If we want to support others heads, we may want to rename it `_head` and rename `model._weights` to `model._head_weights` ?

Reply:
about device : default behavior is to load to cpu, then `model.to(gpu)` , when i try to load directly to device, i sometimes have errors, so I will fall back to mandatory calls to `.to`

Reply:
about _model_weight : indeed there will need some cleanup at some time: we can either load a traced model, some weight to put in _classif head, a prototxt, weight to a model that is defined by a prototxt (and that use the same extension as traced nets), and weight for a native model. We should clean up file extensions in order to be able to load several of them in the future.

Reply:
> about device : default behavior is to load to cpu, then `model.to(gpu)` , when i try to load directly to device, i sometimes have errors, so I will fall back to mandatory calls to `.to`

Can we double check on that ? Like, are we loading the model twice effectively ? I don't want to slow this PR down, but I fee if the PR is called 'load weights only once', let's check on that :)

Reply:
check on what ? do you want to try to load models directly to device ?